rosidl_generator_py/rosidl_generator_py/generate_py_impl.py

```diff
@@ -104,23 +118,46 @@ def generate_py(generator_arguments_file, typesupport_impls):
             import_list['%s  # noqa\n' % type_] = 'from %s.%s._%s import %s\n' % \
                 (args['package_name'], subfolder, module_name, type_)
 
-        with open(os.path.join(args['output_dir'], subfolder, '__init__.py'), 'w') as f:
-            for import_line in sorted(import_list.values()):
-                f.write(import_line)
-            for noqa_line in sorted(import_list.keys()):
-                        f.write(noqa_line)
+        path_to_module = os.path.join(args['output_dir'], subfolder, '__init__.py')
+
+        content = ""
+        if os.path.isfile(path_to_module):
+            with open(path_to_module, 'r') as f:
+                content = f.read()
+        with open(path_to_module, 'w') as f:
+            block_name = args['package_name']
+            if action_specs:
+                block_name += '_action'
+            content = re.sub(
+                r"# BEGIN %s$.*^# END %s" % (block_name, block_name),
+                '', content, 0, re.M | re.S
+            )
```

Comment:
Is this part where it modifies the previously generated `__init__.py` the fragile part you were referring to?

Reply:
That and adding `_action` to the package name whenever necessary (but not everywhere) to avoid name collisions.